test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,171 @@
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayapiclientset "sigs.k8s.io/gateway-api/pkg/client/clientset/versioned"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:GatewayAPIController][Feature:Router][apigroup:gateway.networking.k8s.io]", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithPodSecurityLevel("gatewayapi-controller", admissionapi.LevelBaseline)
+	)
+	const (
+		// The expected OSSM subscription name.
+		expectedSubscriptionName = "servicemeshoperator3"
+		// The expected OSSM operator namespace.
+		expectedSubscriptionNamespace = "openshift-operators"
+		// The gatewayclass name used to create ossm and other gateway api resources.
+		gatewayClassName = "openshift-default"
+		// gatewayClassControllerName is the name that must be used to create a supported gatewayClass.
+		gatewayClassControllerName = "openshift.io/gateway-controller/v1"
+	)
+	g.BeforeEach(func() {
+		// create the default gatewayClass and confirm that it is accepted
+		gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+
+		gatewayClass := buildGatewayClass(gatewayClassName, gatewayClassControllerName)
+		gwc, err := gwapiClient.GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
+		if err != nil {
+			e2e.Logf("Gateway Class %s already exists, or has failed to be created, checking its status", gwc.Name)
+		}
+
+		errCheck := checkGatewayClass(oc, gatewayClassName)
+		o.Expect(errCheck).NotTo(o.HaveOccurred())
+		e2e.Logf("GatewayClass %s successfully installed and accepted!", gwc.Name)
+
+	})
+
+	g.Describe("Verify Gateway API controller resources are created", func() {
+		g.It("and ensure OSSM related resources are created", func() {
+			coreClient := clientset.NewForConfigOrDie(oc.AdminConfig())
+			//check the subscription
+			g.By("Check OSSM subscription and CSV")
+			csvName, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.installedCSV}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			e2e.Logf("The subscription is installed and the CSV is: %v", csvName)
+			csvStatus, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "clusterserviceversion", csvName, "-o=jsonpath={.status.phase}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(csvStatus).To(o.Equal("Succeeded"))
+
+			g.By("check OSSM Pod and istiod pod is present with OSSM Operator installation")
+			podList, err := coreClient.CoreV1().Pods("openshift-operators").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`servicemesh-operator3`))
+
+			podList, err = coreClient.CoreV1().Pods("openshift-ingress").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`istiod-openshift-gateway`))
```

Comment:
Should we check the two pods with ready status?

Reply:
Since the pod is managed by deployment, we could use `Get` deployment/ istiod-openshift-gateway and expect the `status.readyReplicas` is 1. It might be better than the way of listing all pods and checking if the list contains the pod name.